src/tri.rs

```diff
@@ -30,38 +39,44 @@ where
     /// ```
```

Comment:
Could the D::Smaller: Copy bound above be removed (in a future PR, if needed?) It rules out IxDyn for now, is that intentional?

Reply:
This is tricky, and it's more an artifact of the implementation than a fundamental limit of these methods. The use of `Zip::indexed(self.rows())` is where the `D::Smaler: Copy` bound comes from. Fixing this should be possible, but I just have to wrap my head around the arithmetic for how to calculate row number for arbitrarily-strided arrays. It is interesting to note that this is maybe something that would be better fixed by trying to make the underlying `::indexed` call possible on `IxDyn` and then removing the bound here, rather than building out custom iteration logic in this method. Open to suggestions!

Reply:
Oh, I didn't know it came from there. It kind of makes sense now, it's because of indexed. Let's not work around that here.

Reply:
Here is one way to work around it. It's kind of nice since it makes the index logic simpler in the inner loop.

Take index sequence like `[1, 2, 3]` and broadcast up to the space of the rows (like `[[1, 2, 3], [1, 2, 3]]` or something like that.

```diff
         let ncols = self.len_of(Axis(n - 1));
-        Zip::indexed(self.rows())
+        let index_length = self.len_of(Axis(n - 2));
+        let indexes = Array::from_iter(0..index_length);
+        let full_indexes = indexes.broadcast(self.dim.try_remove_axis(Axis(n - 1))).unwrap();
+        Zip::from(full_indexes)
+            .and(self.rows())
             .and(res.rows_mut())
-            .for_each(|i, src, mut dst| {
-                let row_num = i.into_dimension().last_elem();
+            .for_each(|&row_num, src, mut dst| {
                 let mut upper = match k >= 0 {
```

Reply:
I like it and it works! Added it to the PR and removed the trait bound.